### PR TITLE
Fix for #886

### PR DESCRIPTION
**Pull request by haiodo**
Signed-off-by: Andrey Sobolev <haiodo@xored.com>

If Dataplane pods had different nets we need to have routes between them inside vpp.

## Description
Adds proper routes between data planes in case they have different network prefixes.

## Motivation and Context
To support this kind of scenarios.

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by edwarnicke** [comment]
Question... why are we changing the remoteConnection converters etc?  We only need to fix the routes  here for the vppagent-dataplane's Pod interface...

**Comment by edwarnicke** [comment]
Also not clear why we are altering the NSEs around this issue...

File: examples/cmd/vppagent-firewall-nse/vppagent.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func (vxc *vppAgentXConnComposite) crossConnecVppInterfaces(ctx context.Context,
 
 	conversionParameters := &converter.CrossConnectConversionParameters{
 		BaseDir: baseDir,
+		Routes: &converter.ExtraRoutesParameters{},
```
Comment (edwarnicke) [inline examples/cmd/vppagent-firewall-nse/vppagent.go]: Is this an unrelated fix?

File: controlplane/pkg/model/model.go [hunk 2]
```diff
@@ -204,6 +207,7 @@ func (i *impl) DeleteClientConnection(connectionId string) {
 		i.Unlock()
 		return
 	}
+	clientConnection.ConnectionState = ClientConnection_Closed
```
Comment (edwarnicke) [inline controlplane/pkg/model/model.go]: Is this an unrelated fix?

File: examples/cmd/vppagent-firewall-nse/vppagent.go (hunk 1 above)
Comment (haiodo) [reply examples/cmd/vppagent-firewall-nse/vppagent.go]: Thanks. Pushed fix.

File: controlplane/pkg/model/model.go (hunk 2 above)
Comment (haiodo) [reply controlplane/pkg/model/model.go]: This fixes concurrency issue, some tests failed on build.

Comment (haiodo) [reply controlplane/pkg/model/model.go]: Fix for concurrency issue and moved marking of connection is closed into this method.
